lib/screen_controller.dart

```diff
@@ -88,6 +89,9 @@ class ScreenController {
     if (action.initiator != null) {
       dataContext.addInvokableContext('this', action.initiator!);
     }
+    if ( event != null ) {
+      dataContext.addInvokableContext('event', event);
```

Comment:
can you remove null check and always do `.addInvokableContext('event', event ?? {})` instead? It's easier to not having to check for null event in javascript e.g. if (event != null && event.source !=null )

Reply:
But that can actually cause a bunch of issues in javascript. If as an app developer, you think that event is not null and start using its properties, you will get errors and won't know what to do. event.source is a required property so when event is not-null, it has to be there and not-null as well. By making the event an empty object, we are making life harder and causing more issues for the app developer because now they won't just have to check for event != null but every single property of the event - event.source != null, event.data != null.

It's a bad practice and highly error prone to put an empty object there when there actually should be null.

Reply:
@vusters please see my comment above.

Reply:
executeAction can be called from outside a widget, in which case event.source would be null, but you can still pass the data. e.g. page's onLoad.

Reply:
no, the event.source would the page in that case. Event always MUST have a source.

Reply:
why enforcing that event always need a source? a Timer can repetitively executeActions and pass new event data each time, but source is irrelevant. Why force the user to pass a source, especially when they don't need or can't do anything to it?